libs/ledger/src/time_based_transaction_status_cache.cpp

```diff
@@ -0,0 +1,160 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2020 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "network/generics/milli_timer.hpp"
+#include "time_based_transaction_status_cache.hpp"
+
+namespace fetch {
+namespace ledger {
+namespace {
+
+constexpr std::chrono::hours   LIFETIME{24};
+constexpr std::chrono::minutes INTERVAL{5};
+
+using TxStatus = TimeBasedTransactionStatusCache::TxStatus;
+
+}  // namespace
+
+/**
+ * Query the status of a specified transaction
+ *
+ * @param digest The digest of the transaction
+ * @return The status object associated for this transaction
+ */
+TxStatus TimeBasedTransactionStatusCache::Query(Digest digest) const
+{
+  {
+    FETCH_LOCK(mtx_);
+
+    auto const it = cache_.find(digest);
+    if (cache_.end() != it)
+    {
+      return it->second.status;
+    }
+  }
```

Comment:
The outer scope here does nothing.

Reply:
Strictly speaking the mutex is release slightly earlier - this was just a copy of the old code, but I take the larger point that this is not necessary in this case - Done